lib/adapters/xhr.js

```diff
@@ -49,7 +49,20 @@ module.exports = function xhrAdapter(config) {
 
     // Listen for ready state
     request[loadEvent] = function handleLoad() {
-      if (!request || (request.readyState !== 4 && !xDomain)) {
+      if (!request) {
+        return;
+      }
+
+      if (request.readyState === this.HEADERS_RECEIVED && config.maxContentLength > -1 &&
+        request.getResponseHeader('Content-Length') > config.maxContentLength) {
+        reject(createError('maxContentLength size of ' + config.maxContentLength + ' exceeded',
+          config, 'ECONNABORTED', request));
+        request.abort();
+        request = null;
+        return;
+      }
+
+      if (request.readyState !== 4 && !xDomain) {
```

Comment:
How about add clean up?
```javascript
// Clean up request
request = null;
```

Reply:
@hg-pyun  hmm , based on the XMLHttpRequest spec https://xhr.spec.whatwg.org/#handler-xhr-onreadystatechange

readyState might be OPENED , HEADERS_RECEIVED, or LOADING, before DONE , and I think we still need to keep request object before the final state.

Reply:
@cctuan Got it. thanks!